src/app/core/services/config/protocol.service.ts

```diff
@@ -1,21 +1,33 @@
+// tslint:disable: forin
 import { HttpClient } from '@angular/common/http'
 import { Injectable } from '@angular/core'
 
 import {
+  DefaultParticipantAttributeOrder,
   DefaultProtocolBranch,
   DefaultProtocolEndPoint,
+  DefaultProtocolGithubRepo,
   DefaultProtocolPath,
   DefaultQuestionnaireFormatURI,
-  DefaultQuestionnaireTypeURI
+  DefaultQuestionnaireTypeURI,
+  GIT_API_URI
 } from '../../../../assets/data/defaultConfig'
 import { ConfigKeys } from '../../../shared/enums/config'
 import { Assessment } from '../../../shared/models/assessment'
+import { sortObject } from '../../../shared/utilities/sort-object'
 import { LogService } from '../misc/log.service'
 import { RemoteConfigService } from './remote-config.service'
 import { SubjectConfigService } from './subject-config.service'
 
 @Injectable()
 export class ProtocolService {
```

Comment:
Can you please document all the functions in this, otherwise it will be hard to understand for another developer. Also please document the procedure and precedence order for pulling notifications in the README.

Reply:
Okay sure will do.